gapis/api/transform2/control_flow.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transform2
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/api/command_generator"
+	"github.com/google/gapid/gapis/config"
+)
+
+type ControlFlow struct {
+	transforms       []Transform
```

Comment:
This should be in the LinearControlFlowGenerator that is implementing the ControlFlowGenerator interface. The LinearControlFlowGenerator should also not take a slice of Transforms from which it builds transform chains, but it should accept *two* pre-constructed transform chains from the outside, where it was created from.